### PR TITLE
Cleanse the code

### PR DESCRIPTION
**Pull request by WeMayNeverKnow**
This is my first pull request so please understand if I did something wrong

### CODE REVIEW

File: functions/api/jisho.py [hunk 1]
```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
...
+                
                 reading: str = kanji.get('reading')
-                if reading and reading not in readings:
+                if reading not in readings:
                     readings.add(reading)
+                
                 word: str = kanji.get('word')
-
-                if word and word not in words:
+                if word not in words:
```
Comment (taichikuji) [inline functions/api/jisho.py:60]: This is intended, and should not be removed. If removed, code breaks. If you're intending on 'cleaning' this cog, you might probably want to, if possible, get the same result with a different approach.

File: functions/api/jisho.py [hunk 2]
```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
...
         for data in response:
             readings: Set[str] = set()
             words: Set[str] = set()
             kanji_link: Set[str] = set()
-
+            
             for kanji in data['japanese']:
+                
                 reading: str = kanji.get('reading')
-                if reading and reading not in readings:
```
Comment (taichikuji) [inline functions/api/jisho.py:50]: This is intended, and should not be removed. If removed, code breaks. If you're intending on 'cleaning' this cog, you might probably want to, if possible, get the same result with a different approach.

File: functions/api/jisho.py [hunk 3]
```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
...
-
-                if word and word not in words:
+                if word not in words:
                     words.add(word)
-
+            
             for word in words:
                 for res in finditer('[一-龯]', word):
-                    if res and res not in kanji_link:
-                        res = f'[{res.group(0)}](https://jisho.org/search/{res.group(0)}%23kanji)'
```
Comment (taichikuji) [inline functions/api/jisho.py:60]: This is intended, and should not be removed. If removed, code breaks. If you're intending on 'cleaning' this cog, you might probably want to, if possible, get the same result with a different approach.

File: functions/api/jisho.py [hunk 4]
```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
...
-        results = []
         if not stats_json:
             return None
-
+        
+        results = []
+        
         size = len(stats_json)
+        color = self.bot.color
+        jisho_url = self.jisho_url.format("%20".join(keywords.split()))
```
Comment (taichikuji) [inline functions/api/jisho.py:107]: I don't get the need of doing this.

File: functions/api/jisho.py [hunk 5]
```diff
@@ -38,83 +42,103 @@ async def run_error(self, ctx, error):
...
+                        "name": "Meanings",
+                        "value": res['english']
+                    }
+                ],
                 "footer": {"text": f"Page {start}/{size}"}
             }
-            embed = Embed.from_dict(em)
-            results.append(embed)
+            
+            results.append(Embed.from_dict(embed_dict))
```
Comment (taichikuji) [inline functions/api/jisho.py:140]: So... You're separating values into variables on line 103. But here you're doing the opposite? Why? Once again, I don't get it.

**Comment by taichikuji** [comment]
I'm going to close this PR and please ask you to;

* For starters, I'd like the commit itself to not have a typo. You can squash the bad commit with a good one.

* I'd like you to properly test your code since you told me in private that you did not test it at all, and I'm concerned it will break multiple things.

* I'd like you to make multiple commits for different things. Having everything in one single commit is a mess if you don't squash, and I can do the squashing myself if needed after doing the PR.

* I'd like you to push to the dev branch, since if something breaks, I can easily fix it afterwards, then push it to master.

Hope you understand.

- Taichikuji

File: functions/api/jisho.py (hunk 5 above)
Comment (wax911) [reply functions/api/jisho.py:140]: Agreed and nesting function calls with other function calls is not always good practice from a readability stand point

**Comment by wax911** [comment]
Perhaps you could add Codacy for code quality analysis @taichikuji https://www.codacy.com/ and @WeMayNeverKnow you could always create an issue prior to putting in the work for a PR this is typically a good habit when contributing to projects and allows you to discuss what you may want to change with the maintainer and after your suggestions get approved you can start working on a PR 😃

**Pull request closed without merging**